lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -240,21 +240,61 @@ void SessionState::set_current_version(
   if (session_uc) {
     session_uc->is_current_version_updated = true;
     session_uc->updated_current_version    = new_session_version;
+    MLOG(MINFO) << " Current version is " << get_current_version();
   }
-  MLOG(MINFO) << " Current version is " << get_current_version();
 }
+
 /* Add PDR rule to this rules session list */
-void SessionState::insert_pdr(SetGroupPDR* rule) {
-  PdrList_.push_back(*rule);
+void SessionState::insert_pdr(
+    SetGroupPDR* rule, bool crit_add, SessionStateUpdateCriteria* session_uc) {
+  // Check if it already exists
+  int32_t Pdr_index;
+  Pdr_index = get_pdr_index(rule->pdr_id());
+  if (Pdr_index != -1) {
+    // Update the existing value
+    PdrList_.at(Pdr_index) = *rule;
+  } else {
+    // Insert the rule
+    PdrList_.push_back(*rule);
+  }
+  // update criteria to be updated
+  if (crit_add) session_uc->pdrs_to_install.push_back(*rule);
+}
+
+/* method to change the PDR state */
+void SessionState::set_all_pdrs(enum PdrState pdr_state) {
+  for (auto& rule : PdrList_) {
```

Comment:
`PdrList_` -> `pdr_list_` 
I also think this should be `std::unordered_map<std::string, SetGroupPDR>` if we're doing these kinds of search operations. What do you think? If the change is too destructive, we can file an issue and tackle it in a separate PR.

Reply:
PdrList_ is vector of SetGroupPDR and single IMSI could be more than one SetGroupPDRs (For uplink and downlink).
We can use unordered_multimap. I am thinking first all basic PRs should be merged into master then we will do change based on performance and best approaches of data structure logic.